NuRadioReco/modules/RNO_G/channelBlockOffsetFitter.py

```diff
@@ -0,0 +1,257 @@
+
+        Parameters
+        ----------
+        event: Event object | None
+        station: Station
+            The station to add block offsets to
+        offsets: float | array | dict
+            offsets to add to the event. Default: 1 mV
+
+            - if a float, add gaussian-distributed of amplitude ``offsets``
+              to all channels specified;
+            - if an array, the length should be the same as the number
+              of blocks in a single trace, and the entries will be
+              interpreted as the amplitudes of the offsets;
+            - if a dict, the keys should be the channel ids, and each
+              value should contain either a float or an array to add to
+              each channel as specified above.
+
+        channel_ids: list | None
+            either a list of channel ids to apply the offsets to, or
+            None to apply the offsets to all channels in the station
+            (default: None).
+
+        """
+
+        if channel_ids is None:
+            channel_ids = station.get_channel_ids()
+        for channel_id in channel_ids:
+            channel = station.get_channel(channel_id)
+            if isinstance(offsets, dict):
+                add_offsets = offsets[channel_id]
+            elif len(np.atleast_1d(offsets)) == 1:
+                add_offsets = np.random.normal(
+                    0, offsets, (channel.get_number_of_samples() // self.block_size)
+                )
+            else:
+                add_offsets = offsets
+
+            # save the added offsets as a channelParameter
+            if channel.has_parameter(channelParameters.block_offsets):
+                block_offsets_old = channel.get_parameter(channelParameters.block_offsets)
+                channel.set_parameter(channelParameters.block_offsets, block_offsets_old + offsets)
+            else:
+                channel.set_parameter(channelParameters.block_offsets, offsets)
+
+            channel.set_trace(
+                channel.get_trace() + np.repeat(add_offsets, self.block_size),
+                channel.get_sampling_rate()
+            )
+
+    def remove_offsets(self, event, station, mode='fit', channel_ids=None):
+        """
+        Remove block offsets from an event
+
+        Fits and removes the block offsets from an event. The removed
+        offsets are stored in the ``channelParameters.block_offsets``
+        parameter.
+
+        Parameters
+        ----------
+        event: NuRadioReco.framework.event.Event | None
+        station: NuRadioReco.framework.station.Station
+            The station to remove the block offsets from
+        mode: 'fit' | 'approximate' | 'stored'
+            
+            - 'fit' (default): fit the block offsets with a minimizer
+            - 'approximate' : use the first guess from the out-of-band component,
+              without any fitting (slightly faster)
+            - 'stored': use the block offsets already stored in the 
```

Comment:
The 'stored' option is probably superfluous, but I thought maybe one might want to undo the block offsets that have already been removed by e.g. the external calibration, or allow for some other fit to compute the block offsets.

In terms of simulation, right now we don't simulate the block offsets anyway, and because e.g. the timings are different I'm not sure how straightforward it would be to simulate them for the `sim_station` (rather than for the `station`, which is what I've been doing so far). Even if we end up doing this for the `sim_station`, I'd be inclined to leave it to the user to retrieve the simulated parameter from the `sim_station` to keep a cleaner separation between the two.